### PR TITLE
Update tips fixes

### PR DESCRIPTION
**Pull request by swisspol**
The 2nd commit looks like a big diff but it's actually trivial: I only removed the `if (status->msg == NULL)` condition and shift the block to the left.

### CODE REVIEW

**Comment by carlosmn** [comment]
The reflog message shouldn't be passed to `git_push_update_tips()`. Its message is always "update by push", as documented.

**Comment by carlosmn** [comment]
What is the fix in the second commit for? It doesn't specify what was wrong or how this fixes it.

**Comment by swisspol** [comment]
I removed the commit for the reflog message change.

With regards to the `Fixed update_tips callback called on failed pushed references` commit, if you walk through the existing implementation, the update tips callback is called even for failed updates since `fire_callback` is still 1. The code is also doing a bunch of work to later ignore it if the update was a failed one. So rather than adding more complexity to the code with another condition to set `fire_callback` to 0, it's easier to check upfront if the update is a failed one.

**Comment by carlosmn** [comment]
Ok, it looks like some cruft accumulated which means we fire the callback for updates which we did not do. It'd be great if the commit explained this error and the fix itself, rather than putting it in a comment in the PR, which doesn't go into version control.

File: src/push.c [hunk 1]
```diff
@@ -180,6 +180,10 @@ int git_push_update_tips(git_push *push)
 	git_vector_foreach(&push->status, i, status) {
 		int fire_callback = 1;
 
+		/* Skip unsuccessful updates which have empty messages */
```
Comment (carlosmn) [inline src/push.c]: non-empty

**Comment by swisspol** [comment]
Should be good now!

**Comment by swisspol** [comment]
Can this be merged?

**Pull request merged**